### PR TITLE
INTYGFV-16701: Simplify ping and session handling

### PR DESCRIPTION
**Pull request by SamuelTilly**
* Do not call `ping` once session is over.
* Do not log failed `ping` requests unless it contains an error.
* Replace deprecated `AnyAction`.
* Remove unnecessary `startSession` reducer.

### CODE REVIEW

File: apps/minaintyg/src/store/slice/session.slice.ts [hunk 1]
```diff
@@ -27,9 +27,6 @@ const sessionSlice = createSlice({
     reset() {
       return initialState
     },
-    startSession(state) {
-      state.hasSession = true
-    },
```
Comment (SamuelTilly) [inline apps/minaintyg/src/store/slice/session.slice.ts:32]: This was only used for testing, is now replaced with using `getUser` endpoint dispatch

File: apps/minaintyg/src/store/middleware/error.middleware.ts [hunk 2]
```diff
@@ -21,6 +25,10 @@ export const errorMiddleware: ThunkMiddleware<RootState> =
   ({ dispatch, getState }) =>
   (next) =>
   (action) => {
+    if (api.endpoints.getSessionPing.matchRejected(action) && !getState().sessionSlice.errorId) {
+      return next(action)
+    }
```
Comment (SamuelTilly) [inline apps/minaintyg/src/store/middleware/error.middleware.ts:30]: No need to log a rejected ping request unless it contains a `errorId`

File: apps/minaintyg/src/store/middleware/error.middleware.ts [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+const hasMessage = (o: unknown): o is { message: string } => isPlainObject(o) && 'message' in o && typeof o.message === 'string'
+
+function getMessage(action: UnknownAction): string {
+  if (isRejectedEndpoint(action)) {
+    if (action.payload && hasMessage(action.payload.data)) {
+      return action.payload.data.message
+    }
+    if (action.error && action.error.message) {
+      return action.error.message
+    }
```
Comment (SamuelTilly) [inline apps/minaintyg/src/store/middleware/error.middleware.ts:15]: `AnyAction` is deprecated, replacing with `UnknownAction`  and added type guard for `data.message`

File: apps/minaintyg/src/components/ProtectedRoute/ProtectedRoute.tsx [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 import { useGetSessionPingQuery } from '../../store/api'
-import { useGetUserQuery } from '../../store/hooks'
+import { useAppSelector, useGetUserQuery } from '../../store/hooks'
 import { SessionDialog } from '../SessionDialog/SessionDialog'
 
 export function ProtectedRoute({ children }: { children: ReactNode }): React.JSX.Element | null {
   const { isError, isLoading, data: user } = useGetUserQuery()
-  const { data: session } = useGetSessionPingQuery(undefined, {
+  const hasSession = useAppSelector((state) => state.sessionSlice.hasSession)
+  const { data: session } = useGetSessionPingQuery(hasSession ? undefined : skipToken, {
```
Comment (SamuelTilly) [inline apps/minaintyg/src/components/ProtectedRoute/ProtectedRoute.tsx:10]: No need to keep calling ping once the session is over.

**Review by ChrilleSigerhed — APPROVED** [review]
Nice improvements, well done!

**Pull request merged**